cmd/vulcan-aws-subdomain-takeover/awsipranges.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2025 Adevinta
+*/
+
+// Package main contains the main logic to detect subdomain takeovers.
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"maps"
+	"net"
+	"net/http"
+	"os"
+	"slices"
+	"sort"
+	"strings"
+)
+
+const (
+	ipRangesFile = "ip-ranges.json"
+	ipRangesURL  = "https://ip-ranges.amazonaws.com/ip-ranges.json"
+)
+
+// ErrPrefixNotFound means tha a prefix wasn't found in the ip ranges.
+var ErrPrefixNotFound = errors.New("prefix not found")
+
+// AWSIPRanges contains two strings, SyncToken and CreateDate, and a slice of Prefixes.
+type AWSIPRanges struct {
+	SyncToken  string `json:"syncToken"`
+	CreateDate string `json:"createDate"`
+	d          ipRangesRetriever
+	IPPrefixes []IPPrefix `json:"prefixes" json_alias:"ip6_prefixes"`
```

Comment:
What is this json_alias:"ip6_prefixes" ?

Reply:
If you look into the file https://ip-ranges.amazonaws.com/ip-ranges.json, you can see there are two groups of ips, IP4 and IP6. The JSON structure is basically the same, only the key changes. Using the alias you can unmarshal those two JSON structures into the same golang struct type.